src/adapters/ws/server.ts

```diff
@@ -74,6 +74,52 @@ class WebSocketsAdapter extends Adapter {
     })
   }
 
+  private getSecurityReqs() {
```

Comment:
extracting these steps into another class to handle auth related things for http and ws would be great.